apps/balancer-tools/seed.ts

```diff
@@ -523,83 +592,98 @@ DO UPDATE SET chain_id = EXCLUDED.chain_id, updated_at = NOW();
-//       );
-//     }
-//   }
-// }
+async function seedVebalRounds() {
+  logIfVerbose("Seeding veBAL rounds");
+
+  const startDate = new Date("2022-04-14T00:00:00.000Z");
+  let roundNumber = 1;
+
+  while (startDate <= new Date()) {
+    const endDate = new Date(startDate);
+    endDate.setUTCDate(endDate.getUTCDate() + 6);
+    endDate.setUTCHours(23, 59, 59, 999);
+
+    await db.execute(sql`
+      INSERT INTO vebal_rounds (start_date, end_date, round_number)
+      VALUES (${startDate.toISOString()}, ${endDate.toISOString()}, ${roundNumber})
+      ON CONFLICT (round_number) DO NOTHING;
+    `);
+    startDate.setUTCDate(startDate.getUTCDate() + 7);
+    roundNumber++;
+  }
+}
+
+async function seedBalEmission() {
+  const timestamps = await db
+    .selectDistinct({ timestamp: poolSnapshots.timestamp })
+    .from(poolSnapshots);
+
+  for (const { timestamp } of timestamps) {
+    if (!timestamp) continue;
+    try {
+      const weeklyBalEmission = balEmissions.weekly(dateToEpoch(timestamp));
+
+      await db
+        .insert(balEmission)
+        .values({
+          timestamp: timestamp,
+          weekEmission: String(weeklyBalEmission),
+        })
+        .onConflictDoNothing()
+        .execute();
+    } catch (error) {
+      console.error(`${error}`);
+    }
+  }
+
+  // balEmissions.weekly(endAtTimestamp);
+}
+
+async function calculateApr() {
+  // Fee APR
+  logIfVerbose("Seeding Fee APR");
+  await db.execute(sql`
+  INSERT INTO swap_fee_apr (timestamp, pool_external_id, collected_fees_usd, value, external_id)
+  SELECT
+	p1.timestamp,
+	p1.pool_external_id,
+	p1.swap_fees - p2.swap_fees AS collected_fees_usd,
+    CASE 
+        WHEN p1.liquidity = 0 THEN 0
+        ELSE ((p1.swap_fees - p2.swap_fees) * (1 - COALESCE(p1.protocol_swap_fee_cache, 0.5)) / p1.liquidity) * 365 * 100 
+    END AS "value",
+  p1.external_id
+FROM pool_snapshots p1
+LEFT JOIN pool_snapshots p2 
+ON p1. "timestamp" = p2. "timestamp" + INTERVAL '1 day'
+AND p1.pool_external_id = p2.pool_external_id
+ON CONFLICT (external_id) DO NOTHING;
+`);
+
+  // veBAL APR
+  logIfVerbose("Seeding veBAL APR");
+  await db.execute(sql`
+INSERT INTO vebal_apr (timestamp, value, pool_external_id, external_id)
+SELECT DISTINCT
+    ps.timestamp,
+    CASE
+        WHEN ps.liquidity = 0 THEN 0
+        ELSE (52 * (be.week_emission * gs.relative_weight * tp.price_usd) / ps.liquidity) * 100
+    END AS value,
+    ps.pool_external_id,
+    ps.external_id
+FROM pool_snapshots ps
+LEFT JOIN vebal_rounds vr ON ps.timestamp BETWEEN vr.start_date AND vr.end_date
+JOIN gauges g ON g.pool_external_id = ps.pool_external_id
+JOIN gauge_snapshots gs ON g.address = gs.gauge_address
+    AND vr.round_number = gs.round_number
+LEFT JOIN token_prices tp ON tp.token_address = '0xba100000625a3754423978a60c9317c58a424e3d'
+    AND tp.timestamp = ps.timestamp
+LEFT JOIN bal_emission be ON be.timestamp = ps.timestamp
+ON CONFLICT (external_id) DO NOTHING;
+`);
```

Comment:
veBAL APR will always be a little different from what is showing on Balancer's UI because the only consider the staked tokens and we consider the whole liquidity @ribeirojose 

I discussed this with @mendesfabio  and the gaugeTotalSupply (the staked tvl) is not saved by timestamp in the subgraph, that's why we are using this approximation. We concluded the values are accurate if they are within the stacking APR range, closer to the min staked APR